src/NHibernate/Impl/AbstractQueryImpl.cs

```diff
@@ -651,6 +651,8 @@ public IQuery SetEnum(string name, Enum val)
 			return this;
 		}
 
+		// Since 5.3
+		[Obsolete("This method was never surfaced to a query interface. Use the overload or extension method taking a generic dictionary instead.")]
 		public IQuery SetProperties(IDictionary map)
```

Comment:
I have considered adding an extension method for it too, but this was causing an ambiguous method call when using a `Dictionary<string, object>`, since it does implement both `IDictionary` and `IDictionary<string, object>`. So it looks to me not worth keeping this method.